src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelper.cs

```diff
@@ -50,6 +53,7 @@ public class HtmlHelper : IHtmlHelper, ICanHasViewContext
             HtmlEncoder = htmlEncoder;
             UrlEncoder = urlEncoder;
             JavaScriptStringEncoder = javaScriptStringEncoder;
+            _logger = loggerFactory.CreateLogger<HtmlHelper>();
```

Comment:
Yeah get the logger directly through DI in the ctor, don't get the factory to create it manually.